### PR TITLE
feat(api): Update Project PUT+GET+DELETE

### PR DESCRIPTION
**Pull request by schew2381**
Update Project APIs to use drf-spectacular. Planning on adding a lot more to Project PUT in future PRs.

Old Put:
Screenshot 2023-06-30 at 1 56 49 PM

New Put:
Screenshot 2023-07-05 at 10 44 07 AM
Screenshot 2023-06-30 at 3 38 30 PM

Old Get:
Screenshot 2023-06-30 at 12 46 21 PM

New Get:
Screenshot 2023-06-30 at 1 55 46 PM

Old Delete:
Screenshot 2023-06-28 at 11 03 29 AM

New Delete (not much difference tbh):
Screenshot 2023-06-28 at 11 03 52 AM

### CODE REVIEW

File: src/sentry/api/endpoints/organization_teams.py [hunk 1]
```diff
@@ -160,7 +160,7 @@ def should_add_creator_to_team(self, request: Request):
         operation_id="Create a New Team",
         parameters=[
             GlobalParams.ORG_SLUG,
-            GlobalParams.name("The name of the team.", required=True),
+            GlobalParams.name("The name for the team.", required=True),
```
Comment (schew2381) [inline src/sentry/api/endpoints/organization_teams.py:163]: nit: standardizing the phrase `The name for the {resource}`

File: src/sentry/api/endpoints/project_details.py [hunk 2]
```diff
@@ -394,26 +403,30 @@ def get(self, request: Request, project: Project) -> Response:
...
-        :param boolean isBookmarked: in case this API call is invoked with a
-                                     user context this allows changing of
-                                     the bookmark flag.
-        :param int digestsMinDelay:
-        :param int digestsMaxDelay:
-        :auth: required
+        Update various attributes and configurable settings for the given project.
+
+        Note that soley having the **`project:read`** scope restricts updatable settings to
+        `isBookmarked` only.
```
Comment (schew2381) [inline src/sentry/api/endpoints/project_details.py:429]: We have a weird permission scope check where we allow `project:read` to hit the endpoint, but restrict it from updating almost every field except `isBookmarked` and `isSubscribed`. AFAIK `isSubscribed` is irrelevant/useless so I excluded it. I'm not sure why this was allowed is the first place but I'm sure there was a reason so I'm afraid that removing `project:read` will break something in the FE

File: src/sentry/api/endpoints/team_projects.py [hunk 3]
```diff
@@ -120,7 +120,7 @@ def get(self, request: Request, team) -> Response:
         parameters=[
             GlobalParams.ORG_SLUG,
             GlobalParams.TEAM_SLUG,
-            GlobalParams.name("The name of the project.", required=True),
+            GlobalParams.name("The name for the project.", required=True),
```
Comment (schew2381) [inline src/sentry/api/endpoints/team_projects.py:123]: nit: standardizing the phrase The name for the {resource}

File: src/sentry/apidocs/parameters.py [hunk 4]
```diff
@@ -336,8 +344,8 @@ class ProjectParams:
         required=False,
         type=build_typed_list(OpenApiTypes.STR),
         description="""
-A list specifying which legacy browser filters should be active. Anything excluded from the list
-will be turned off. The options are:
+Specifies which legacy browser filters should be active. Anything excluded from the list will be
+disabled. The options are:
```
Comment (schew2381) [inline src/sentry/apidocs/parameters.py:348]: nit: Better msg here

**Review by cathteng — COMMENTED** [review]
looks good but i would get another pair of eyes on this 👏

File: src/sentry/api/serializers/models/project.py [hunk 5]
```diff
@@ -825,6 +827,72 @@ def _get_project_to_release_version_mapping(
     }
 
 
+class Plugins(TypedDict):
```
Comment (sentaur-athena) [inline src/sentry/api/serializers/models/project.py:830]: Plugin singular?

File: src/sentry/apidocs/examples/project_examples.py [hunk 6]
```diff
@@ -110,6 +110,324 @@
...
+    "hasProfiles": True,
+    "hasReplays": False,
+    "hasSessions": True,
+    "isInternal": True,
+    "isPublic": False,
+    "avatar": {"avatarType": "letter_avatar", "avatarUuid": None},
+    "color": "#3fbf67",
+    "status": "active",
+    "team": {"id": "2", "name": "Powerful Abolitionist", "slug": "powerful-abolitionist"},
+    "teams": [{"id": "2", "name": "Powerful Abolitionist", "slug": "powerful-abolitionist"}],
```
Comment (sentaur-athena) [inline src/sentry/apidocs/examples/project_examples.py:159]: Not related to your code changes, but what is the difference between team and teams?

File: src/sentry/apidocs/examples/project_examples.py [hunk 7]
```diff
@@ -110,6 +110,324 @@
...
+            "resourceLinks": [
+                {"title": "Report Issue", "url": "https://github.com/getsentry/sentry/issues"},
+                {
+                    "title": "View Source",
+                    "url": "https://github.com/getsentry/sentry/tree/master/src/sentry_plugins",
+                },
+            ],
+        }
+    ],
+    "platforms": ["native", "other", "python"],
```
Comment (sentaur-athena) [inline src/sentry/apidocs/examples/project_examples.py:413]: Again not related and just asking for my info. What is platform "other"? I don't see it on UI

File: src/sentry/apidocs/parameters.py [hunk 8]
```diff
@@ -305,6 +305,14 @@ class ProjectParams:
         description="Activate or deactivate the client key.",
     )
 
+    IS_BOOKMARKED = OpenApiParameter(
+        name="isBookmarked",
+        location="query",
+        required=False,
+        type=bool,
+        description="Enables setting the bookmark flag when the request is invoked with a user context.",
```
Comment (sentaur-athena) [inline src/sentry/apidocs/parameters.py:313]: Is this user facing? If so let's make it more user friendly, like: bookmarks a project ?

File: src/sentry/apidocs/examples/project_examples.py (hunk 6 above)
Comment (schew2381) [reply src/sentry/apidocs/examples/project_examples.py:159]: Teams can have multiple listed, while Team is always a single team. I'm not too sure what the difference is, but it's what's currently returned by the API and was previously defined in our schema like this

File: src/sentry/apidocs/examples/project_examples.py (hunk 7 above)
Comment (schew2381) [reply src/sentry/apidocs/examples/project_examples.py:413]: We have the key `platform` defined at the beginning of the example that's only what the user selected on the UI. I'm not sure how these other platforms get added to 'platforms' or what `other` is unfortunately

File: src/sentry/apidocs/parameters.py (hunk 8 above)
Comment (schew2381) [reply src/sentry/apidocs/parameters.py]: Changed to `Enables starring the project within the projects tab.`

File: src/sentry/apidocs/examples/project_examples.py (hunk 6 above)
Comment (sentaur-athena) [reply src/sentry/apidocs/examples/project_examples.py:159]: backward compatibility maybe? teams should include team anyways so I suggest removing team (not here, another PR). @leedongwei do you have any thoughts on if this change would be safe?

**Review by sentaur-athena — APPROVED** [review]

**Pull request merged**